### PR TITLE
fix(submodules): make git submodule update always recursive

### LINKED ISSUE #30294: Support recursive submodules
### Describe the proposed change(s).

Discussion: https://github.com/renovatebot/renovate/discussions/30189

Reproduction: https://github.com/Josef-Reichardt/renovate-bot-reproduction-recursive-submodules/

### PR DESCRIPTION
**Pull request by Josef-Kainz-Reichardt**
## Changes

Added `--recursive` to the git submodules update command.

## Context

Discussion: #30189

Closes #30294

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/configuration-options.md [hunk 1]
```diff
@@ -442,7 +442,7 @@ For `sbt` note that Renovate will update the version string only for packages th
 
 ## cloneSubmodules
 
-Enabling this option will mean that any detected Git submodules will be cloned at time of repository clone.
+Enabling this option will mean that any detected Git submodules will be cloned at time of repository clone. Submodules are always cloned recursively.
```
Comment (rarkins) [inline docs/usage/configuration-options.md:445]:
```suggestion
Enabling this option will mean that any detected Git submodules will be cloned at time of repository clone.
Submodules are always cloned recursively.
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Have you tested this on a real repo?

**Comment by Josef-Reichardt** [comment]
> Have you tested this on a real repo?

Yes, I don't know how to run my branch as GitHub action, so  I just startet it via `pnpm start ...` against a repository in our private GitLab.

**Review by rarkins — DISMISSED** [review]

**Comment by rarkins** [comment]
Awaiting v38 release before merging

**Review by viceice — COMMENTED** [review]
needs prettier fix

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**